src/libos/src/vm/vm_manager.rs

```diff
@@ -74,14 +74,20 @@ impl VMManager {
         match addr {
             VMMapAddr::Any => {}
             VMMapAddr::Hint(addr) => {
-                let target_range = unsafe { VMRange::from_unchecked(addr, addr + size) };
+                let end = addr
+                    .checked_add(size)
+                    .ok_or_else(|| errno!(ENOMEM, "end address overflow"))?;
```

Comment:
Maybe the errno should be EOVERFLOW ?

Reply:
From man page, EOVERFLOW implies it is happened on 32-bit architecture.
```
On 32-bit architecture together with the large file extension (i.e., using 64-bit off_t): the number of pages used for length plus number of pages used for offset would overflow unsigned long (32 bits).
```
ENOMEM is also the LTP test cases expected error number.
```
The process's maximum number of mappings would have been exceeded.
```